src/react/providers/room-provider.tsx

```diff
@@ -0,0 +1,79 @@
+import { RoomOptions, RoomOptionsDefaults } from '@ably/chat';
+import React, { ReactNode, useEffect, useMemo } from 'react';
+
+import { RoomContext } from '../contexts/room-context.js';
+import { useChatClient } from '../hooks/use-chat-client.js';
+
+/**
+ * Props for the RoomProvider component.
+ *
+ * Only `id` (the room ID) is required, all other props are optional.
+ */
+export interface RoomProviderProps {
+  /** Required. The id of the room. */
+  id: string;
+
+  /**
+   * Optional. The options to use when creating the room. If none are provided,
+   * the default is {@link RoomOptionsDefaults}.
+   */
+  options?: RoomOptions;
```

Comment:
In the base client, we don't allow options to be optional, you have to be explicit about using `RoomOptionsDefaults`, should we use the same here?

Reply:
I thought if you just want the default/just play around it's easier to simply `<RoomProvider id={"abc"}>`.

Happy with either way

Reply:
thanks both, made it required